values/dev-stack-values.yaml

```diff
@@ -1 +1,4 @@
 extraPorts: ###ZARF_VAR_NGINX_EXTRA_PORTS###
+
+coreDnsOverrides: |
+  ###ZARF_VAR_COREDNS_OVERRIDES###
```

Comment:
Can we drop this templated ZARF_VAR and use Zarf helm overrides?

Reply:
Did you mean UDS helm overrides?

In my experience, k3d-core-slim-dev is usually deployed without a uds-config.yaml, and we've been using `--set` for any Zarf variables we need to manipulate, like `K3D_EXTRA_ARGS`.

Reply:
Without, meaning we usually don't supply one when we use it for LeapfrogAI dev purposes.

We usually build the GPU dev cluster with the following Make target snippets:
```make
build-k3d-gpu:
	@cd ${MAKEFILE_DIR} && \
	docker build \
	${DOCKER_FLAGS} \
	--platform linux/amd64 \
	-t ghcr.io/defenseunicorns/leapfrogai/k3d-gpu:${LOCAL_VERSION} .

create-uds-gpu-cluster: build-k3d-gpu
	@uds deploy ${UDS_VERSION} \
	${ZARF_FLAGS} \
	--set K3D_EXTRA_ARGS="--gpus=all \
	--image=ghcr.io/defenseunicorns/leapfrogai/k3d-gpu:${LOCAL_VERSION}" --confirm
```

Reply:
So Zarf also has alpha functionality to create Helm override variables - so you could something like:
```
    charts:
      - name: metallb
        namespace: uds-dev-stack
        url: https://metallb.github.io/metallb
        version: 0.14.8
      - name: uds-dev-stack
        namespace: uds-dev-stack
        localPath: chart
        version: 0.2.0
        valuesFiles:
          - "values/dev-stack-values.yaml"
        variables:
          - name: COREDNS_OVERRIDES
            description: "Core DNS default overrides"
            path: coreDnsOverrides
```

The advantage would be you could rely on the default values that are in chart's `values.yaml` instead of having to template the dev-stack-values.yaml with `###ZARF_VAR...###` only to then redefine the same default if that makes sense.

Reply:
Oh wow, that is a cool new feature. I've been following the pattern UDS Core has with some of the Zarf variables versus overrides, and it's been a bit confusing and I thought I finally got it down. This new feature makes it way easier to consolidate it down to the Zarf level instead (which is way better IMO) of the bundle config level.

I'll play around and try to implement this tomorrow!

Reply:
What version of Zarf does this alpha feature exist? And is it already on the docs pages? Is there a lot of community support behind this alpha feature?

Reply:
It was released in Zarf 0.33.0 via https://github.com/zarf-dev/zarf/pull/2403.  As far as I can tell it is only referenced in this example: https://docs.zarf.dev/ref/examples/helm-charts/#zarfyaml but [the ADR](https://github.com/zarf-dev/zarf/blob/main/adr/0023-chart-override.md) has a bit of context as well.

I can't really speak to the broader Zarf community support but I know at least one community member that is excited about this feature (me).

Reply:
https://docs.zarf.dev/ref/values/#helm-chart-mapping also has some docs on it.

Reply:
I implemented the Zarf helm overrides (commit: 0840a69)  but I am seeing an odd artifact in the rendered chart that did not exist using the standard Zarf variable pattern (commit: b2316a5). Everything is indented/spaced as-is from the rendered template in-cluster.

```yaml
  uds.override: |2

        rewrite {
          name regex (.*\.admin\.uds\.dev) admin-ingressgateway.istio-admin-gateway.svc.cluster.local answer auto
        }
        rewrite {
          name regex (.*\.uds\.dev) tenant-ingressgateway.istio-tenant-gateway.svc.cluster.local answer auto
        }
```

Reply:
It seems to be rendering the multi-line string incorrectly.

Reply:
Due to this issue: https://github.com/zarf-dev/zarf/issues/2783, I won't be using Zarf helm overrides for this PR. A refactor can be made later on.

Reply:
I think your issue might just be in your helm template.  Looking at the `ConfigMap/coredns-custom` when you run `helm template .` in the chart you see the same `|2`.

I believe updating the ConfigMap template to use `indent` instead of `nindent` it should fix it.

Reply:
Ah I see, the typo was affecting it. Thank you, I will test it again. Been juggling a lot of other PRs/branches so I must be getting fatigued haha

Reply:
Implemented!